samples/backtest_grid.py

```diff
@@ -0,0 +1,329 @@
+        @param martingale_sizes: True to increase position sizes in grid
+        @param borrowing_disabled: Disable loans on platform (ie: do not use short trades), default= False
+        """
+        assert position_amount > 0
+
+        # configuration
+        self._exchange = exchange
+        self._position_amount = position_amount
+        self._grid_size = grid_size
+        self._grid_step = grid_step
+        self._hedging = hedging
+        self._martingale_sizes = martingale_sizes
+        self._borrowing_disabled = borrowing_disabled
+
+        # storage lists
+        self._positions: Dict[bs.Pair, PositionInfo] = {}
+        self._buy_orders_ids: List[str] = []
+        self._sell_orders_ids: List[str] = []
+
+    async def on_trading_signal(self, trading_signal: BaseTradingSignal):
+        """
+        Handle trading signals
+
+        @param trading_signal: TradingSignal instance
+        """
+        pairs = list(trading_signal.get_pairs())
+        logging.info(StructuredMessage("Trading signal", pairs=pairs))
+
+        try:
+            coros = []
+            for pair, target_position in pairs:
+                if self._borrowing_disabled and target_position == bs.Position.SHORT:
+                    target_position = bs.Position.NEUTRAL
+                coros.append(self.switch_position(pair, target_position))
+            await asyncio.gather(*coros)
+        except Exception as e:
+            logging.exception(e)
+
+    async def switch_position(self, pair: bs.Pair, target_position: bs.Position, force: bool = False):
+        """
+        Handle change in trading position (SHORT, LONG, NEUTRAL)
+
+        @param pair: Pair instance
+        @param target_position: New position
+        @param force: True if user want to execute even if old position == new position
+        @return:
+        """
+        current_pos_info = await self.get_position_info(pair)
+
+        # Unless force is set, we can ignore the request if we're already there.
+        if not force and any([
+            # current_pos_info is None and target_position == bs.Position.NEUTRAL,
+            (
+                    current_pos_info is not None
+                    and current_pos_info == target_position
+            )
+        ]):
+            return
+        self._positions[pair] = PositionInfo(pair, target_position)
+
+        if target_position != bs.Position.NEUTRAL:
+            return
+
+        # SIGNAL NEUTRAL (BBands cross the middle)
+        buy_order_ids = [order.id for order in await self._exchange.get_open_orders(pair) if
+                         order.operation == OrderOperation.BUY]
+
+        if len(buy_order_ids) == 0:
+
+            (bid, ask), pair_info = await asyncio.gather(
+                self._exchange.get_bid_ask(pair),
+                self._exchange.get_pair_info(pair),
+            )
+            order_size = bs.truncate_decimal(Decimal(self._position_amount), pair_info.base_precision)
```

Comment:
This is assuming that self._position_amount was set it base units. Based on the class docs:
```
@param position_amount: Size of lot or amount of quote value to trade
```

Reply:
ok, will fix it.